lib/json_api_client/resource.rb

```diff
@@ -480,27 +486,39 @@ def reset_request_select!(*resource_types)
 
     protected
 
-    def method_missing(method, *args)
-      association = association_for(method)
+    def relationship_definition_for(name)
+      relationships[name] if relationships && relationships.has_attribute?(name)
+    end
 
-      return super unless association || (relationships && relationships.has_attribute?(method))
+    def included_data_for(name, relationship_definition)
+      last_result_set.included.data_for(name, relationship_definition)
+    end
 
-      return nil unless relationship_definitions = relationships[method]
+    def relationship_data_for(name, relationship_definition, association)
+      return unless relationship_definition
 
       # look in included data
-      if relationship_definitions.key?("data")
-        return last_result_set.included.data_for(method, relationship_definitions)
+      if relationship_definition.key?("data")
+        return included_data_for(name, relationship_definition)
       end
 
-      if association = association_for(method)
-        # look for a defined relationship url
-        if relationship_definitions["links"] && url = relationship_definitions["links"]["related"]
-          return association.data(url)
-        end
+      url = relationship_definition["links"]["related"]
+      if relationship_definition["links"] && url
+        return association.data(url)
       end
+
       nil
     end
 
+    def method_missing(method, *args)
+      association = association_for(method)
+      relationship_definition = relationship_definition_for(method)
```

Comment:
since creating an association generates the accessor methods, will a relationship ever get into `method_missing`? 
If not, do we need this code?

Reply:
good catch)

Reply:
@gaorlov, I've changed it. Take a look

Reply:
Looks great! Nicely done.